locust/configuration.py

```diff
@@ -79,17 +78,96 @@ def update_json_config(self, json_added, json_path, options, list_column, config
         else:
             json_final = json_added
         jsonpath_expr = parse(json_path)
-
         matches = jsonpath_expr.find(data)
-        
+
         if len(matches)==0:
-            return make_response(json.dumps({'success':False, 'message':'JSON path not found.'}))
+            return make_response(json.dumps({'success':False, 'message':'JSON path not found.', 'key_pass_status':False}))
         
         for match in matches:
             data = ClientConfiguration.update_json(data, ClientConfiguration.get_path(match), json_final)
         
-        return make_response(json.dumps({'success':True, 'data':json.dumps(data, indent=4)}))
+        if 'data' in kwargs:
+            return data
+        else:
+            return make_response(json.dumps({'success':True, 'data':json.dumps(data, indent=4), 'key_pass_status':False}))
+
+    def update_json_each_row_depend_on_key(self, json_added, json_path, options, list_column, config_text, key_json, key_csv):
+        """
+        Write JSON file configuration each data json array and csv row depends on a key
+        """ 
+        temppath = json_path.split('.')
+        if(len(temppath[len(temppath)-2]) > 0):
+            return make_response(json.dumps({'success':False, 'message':'Check again your jsonpath'}))
+        
+        data = literal_eval(config_text)
+        
+        for x in xrange(1,len(temppath)-1):
+            if(len(temppath[x]) > 0):
+                data = data[temppath[x]]
         
+        datas = None
+        missing_data_key_csv = []
+        for y in xrange(0,len(json_added)):
+            index_search = ClientConfiguration.find_index(data, key_json, json_added[y][key_csv])
+            if index_search < 0:
+                missing_data_key_csv.append(json_added[y][key_csv])
+            else:
+                json_path_iter = '.'.join(temppath[:-1]) + "[" + str(index_search) + "]." + temppath[-1]
+                if y==0:
```

Comment:
a little tips, you can use `data_each_iter = literal_eval(config_text) if y==0 else datas` for shorthand